ChattoAdditions/Source/Chat Items/CompoundMessage/CompoundMessagePresenter.swift

```diff
@@ -67,60 +70,96 @@ public final class CompoundMessagePresenter<ViewModelBuilderT, InteractionHandle
-        return collectionView.dequeueReusableCell(withReuseIdentifier: .compoundCellReuseId,
-                                                  for: indexPath)
+    open override func dequeueCell(collectionView: UICollectionView, indexPath: IndexPath) -> UICollectionViewCell {
+        let cellReuseIdentifier = self.compoundCellReuseId
+        collectionView.register(CompoundMessageCollectionViewCell.self, forCellWithReuseIdentifier: cellReuseIdentifier)
+        return collectionView.dequeueReusableCell(withReuseIdentifier: cellReuseIdentifier, for: indexPath)
     }
 
-    public override func heightForCell(maximumWidth width: CGFloat,
+    open override func heightForCell(maximumWidth width: CGFloat,
                                        decorationAttributes: ChatItemDecorationAttributesProtocol?) -> CGFloat {
         let layoutConstants = self.cellStyle.layoutConstants(viewModel: self.messageViewModel)
         let maxWidth = (width * layoutConstants.maxContainerWidthPercentageForBubbleView)
         return self.layoutProvider.layout(forMaxWidth: maxWidth).size.height
     }
 
-    public override func configureCell(_ cell: BaseMessageCollectionViewCell<CompoundBubbleView>,
+    open override func configureCell(_ cell: BaseMessageCollectionViewCell<CompoundBubbleView>,
                                        decorationAttributes: ChatItemDecorationAttributes,
                                        animated: Bool,
                                        additionalConfiguration: (() -> Void)?) {
-        guard let compoundCell = cell as? CompoundMessageCollectionViewCell<ModelT> else {
-            assertionFailure("\(cell) is not CompoundMessageCollectionViewCell<\(ModelT.self)>")
+        guard let compoundCell = cell as? CompoundMessageCollectionViewCell else {
+            assertionFailure("\(cell) is not CompoundMessageCollectionViewCell")
             return
         }
 
         super.configureCell(compoundCell, decorationAttributes: decorationAttributes, animated: animated) { [weak self] in
             defer { additionalConfiguration?() }
             guard let sSelf = self else { return }
-            guard compoundCell.lastAppliedConfiguration != sSelf.messageModel else { return }
-            compoundCell.lastAppliedConfiguration = sSelf.messageModel
-            let modules = sSelf.contentFactories.map { $0.createMessageModule(forModel: sSelf.messageModel) }
+
             let bubbleView = compoundCell.bubbleView!
             bubbleView.viewModel = sSelf.messageViewModel
-            bubbleView.style = sSelf.compoundCellStyle
-            bubbleView.decoratedContentViews = modules.map { .init(module: $0) }
             bubbleView.layoutProvider = sSelf.layoutProvider
-            bubbleView.accessibilityIdentifier = sSelf.accessibilityIdentifier
+            bubbleView.style = sSelf.compoundCellStyle
+            bubbleView.setNeedsLayout()
```

Comment:
Why this call is needed here? Shouldn't it be the responsibility of each concrete view to update its own layout based on state changes?

Reply:
Fixed.